### PR TITLE
feat: Added Modal component

### PR DESCRIPTION
**Pull request by arjunbharti**
Created _modal component_ with heading, model body and modal footer with two buttons.

### CODE REVIEW

File: styles/utils.css [hunk 1]
```diff
@@ -17,5 +17,98 @@
     --primary-button-hover: rgb(21, 50, 180);
     --secondary-button-hover: rgb(236, 233, 233);
     --secondary-hover: rgb(220, 219, 245);
```
Comment (ark2002) [inline styles/utils.css]: dont mix  rgb and hex

**Review by ark2002 — CHANGES_REQUESTED** [review]
just a single change rest looks fine

**Review by SHIKHASINGH1506 — APPROVED** [review]

File: components/modal.css [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    display: none;
+    position: fixed;
+    z-index: 1;
+    left: 0;
+    top: 0;
+    width: 100%;
+    height: 100%;
+    overflow: auto;
+    background-color: rgb(0,0,0);
+    background-color: rgba(0,0,0,0.4);
```
Comment (bishnoimukesh) [inline components/modal.css:19]: use var for color like you did above for consistency

File: styles/utils.css (hunk 1 above)
Comment (bishnoimukesh) [inline styles/utils.css]: either or hex code or name or RGB. don't mix all of them

**Review by bishnoimukesh — COMMENTED** [review]
make the navigation links workable
some changes are suggested to look rest all good to go.

File: styles/utils.css (hunk 1 above)
Comment (arjunbharti) [reply styles/utils.css]: Noted, will change at time end

Comment (arjunbharti) [reply styles/utils.css]: Noted, will update at the end else there will be merge conflicts.

**Pull request merged**